feat(workflow): add relations panel to visualize dependencies

Minamiyama opened the pull request:
## Summary

Adds a Relations tab to the workflow editor that parses variable references and displays each node’s dependencies and dependents in real time, making navigation and debugging easier.

fixed #22004
close https://github.com/langgenius/dify/issues/22609

## Screenshots


https://github.com/user-attachments/assets/0e73ee2d-419f-4cef-9c1c-f138623f7aac


### Update: dim nodes, and show data flowing effect of relative nodes


https://github.com/user-attachments/assets/52fadf6d-51a5-41de-86a6-c53a9f38c73c



## Checklist

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

Minamiyama committed:
feat(workflow): add node relations panel to display dependencies

Introduce a new relations tab in workflow panel to visualize node dependencies and dependents. Includes:
- New RelationType enum for dependency types
- Internationalized UI text for multiple languages
- Components for displaying relations with arrows and empty states
- Logic to determine and display dependency relationships

Minamiyama committed:
feat(workflow): add node relations panel to display dependencies

Introduce a new relations tab in workflow panel to visualize node dependencies and dependents. Includes:
- New RelationType enum for dependency types
- Internationalized UI text for multiple languages
- Components for displaying relations with arrows and empty states
- Logic to determine and display dependency relationships

crazywoola commented:
Thanks for the contribution, please link an existing issue in the description.

Minamiyama commented:
> Thanks for the contribution, please link an existing issue in the description.

done

crazywoola commented:
Hello, please add my wechat crazyphage. Not sure if I have sent this message before.

Minamiyama committed:
fix(workflow-panel): pass relationType prop to all Container instances

Ensure consistent behavior by passing relationType prop to Container components in both conditional rendering cases

Minamiyama committed:
fix(workflow-panel): pass relationType prop to all Container instances

Ensure consistent behavior by passing relationType prop to Container components in both conditional rendering cases

Minamiyama committed:
Merge branch 'main' into feat/relations

Minamiyama committed:
Merge branch 'main' into feat/dependency-relations

# Conflicts:
#	web/app/components/workflow/nodes/_base/components/workflow-panel/index.tsx

Minamiyama committed:
Merge branch 'feat/relations' into feat/dependency-relations

# Conflicts:
#	web/app/components/workflow/nodes/_base/components/workflow-panel/index.tsx

Minamiyama committed:
feat(workflow): add node dimming and temporary edges for dependency visualization

- Implement dimming of non-related nodes when Shift key is pressed to highlight dependencies
- Add temporary dashed edges to visualize connections between selected node and its dependencies
- Introduce _dimmed and _isTemp flags in node and edge types to control visual states
- Add keyboard shortcuts (Shift key) to trigger dimming and undimming functionality

crazywoola committed:
fix: remove relations pannel

crazywoola committed:
chore: add Template

crazywoola committed:
chore: add Template

crazywoola committed:
Merge branch 'main' into feat/dependency-relations

DavideDelbianco commented:
I cannot see the "tab" making this feature not so mobile-friendly
also keeping SHIFT pressed for long, makes the assistive technologies in windows popup-up asking if i want to enable caps lock
